### PR TITLE
More clean code, add typehints, fix phpdocs, add new methods.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/db/pull/178

### CODE REVIEW

File: src/ActiveDataProvider.php [hunk 1]
```diff
@@ -50,7 +51,7 @@
  * For more details and usage information on ActiveDataProvider, see the
  * [guide article on data providers](guide:output-data-providers).
  */
-class ActiveDataProvider
+class ActiveDataProvider extends DataReaderProvider
```
Comment (samdark) [inline src/ActiveDataProvider.php:54]: How about making it similar to interface?

```suggestion
class ActiveDataReader extends DataReaderProvider
```

File: src/ActiveDataProvider.php [hunk 2]
```diff
@@ -126,7 +127,7 @@ public function prepareQuery(): QueryInterface
             $query->limit($pagination->getLimit())->offset($pagination->getOffset());
         }
         if (($sort = $this->getSort()) !== false) {
-            $query->addOrderBy($sort->getOrders());
+            //$query->addOrderBy($sort->getOrder());
```
Comment (samdark) [inline src/ActiveDataProvider.php:130]: Doesn't look alright.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ActiveDataProvider.php (hunk 2 above)
Comment (terabytesoftw) [reply src/ActiveDataProvider.php]: Fixed.

File: src/ActiveDataProvider.php (hunk 1 above)
Comment (terabytesoftw) [reply src/ActiveDataProvider.php]: Fixed

**Review by samdark — APPROVED** [review]

**Pull request merged**